hexkit/utils.py

```diff
@@ -15,47 +15,18 @@
 
 """General utilities that don't require heavy dependencies."""
 
-import os
-import signal
-from pathlib import Path
-from typing import Callable, Optional
 
-TEST_FILE_DIR = Path(__file__).parent.resolve() / "test_files"
+class NonAsciiStrError(RuntimeError):
+    """Thrown when non-ASCII string was unexpectedly provided"""
 
-TEST_FILE_PATHS = [
-    TEST_FILE_DIR / filename
-    for filename in os.listdir(TEST_FILE_DIR)
-    if filename.startswith("test_") and filename.endswith(".yaml")
-]
+    def __init__(self, str_value: str):
+        """Prepare custom message."""
+        super().__init__(f"Non-ASCII string provided: {str_value}")
```

Comment:
```suggestion
        super().__init__(f"Non-ASCII string provided: {str_value!r}")
```
Better to use the repr() of the string in the message, particularly when it's not ASCII.

Reply:
Great comment, thanks